### PR TITLE
[WIP] Feat/linstor clone

### PR DESCRIPTION
**Pull request by Wescoeur**

### CODE REVIEW

File: drivers/linstorvolumemanager.py [hunk 1]
```diff
@@ -278,6 +283,27 @@ def metadata(self, metadata):
...
+            filter_by_stor_pools=[self._group_name]
+        ).storage_pools
+
+        disconnected_hosts = set()
+        for pool in pools:
+            for report in pool.reports:
+                if report.ret_code & linstor.consts.WARN_NOT_CONNECTED == \
+                        linstor.consts.WARN_NOT_CONNECTED:
+                    disconnected_hosts.add(pool.node_name)
+                    continue
```
Comment (benjamreis) [inline drivers/linstorvolumemanager.py]: Is the `continue` necessary here since it's the end of the loop?

File: drivers/LinstorSR.py [hunk 2]
```diff
@@ -943,12 +984,60 @@ def _update_device_name(self, device_name):
...
+        if self.vdi_type == vhdutil.VDI_TYPE_RAW:
+            self.size = volume_info.virtual_size
+        else:
+            self.size = vhdutil.getSizeVirt(snap_path)
+        self.utilisation = volume_info.physical_size
+
+        # 5. Update sm config.
+        snap_vdi.sm_config = {}
+        snap_vdi.sm_config['vdi_type'] = vhdutil.VDI_TYPE_VHD
+        snap_vdi.sm_config['vhd-parent'] = snap_parent
```
Comment (benjamreis) [inline drivers/LinstorSR.py]:
```py
snap_vdi.sm_config = {
    'vdi_type': vhdutil.VDI_TYPE_VHD,
    'vhd-parent': snap_parent,
}
```

Also why `vdi_type` uses underscore and `vhd-parent` uses dash?

File: drivers/linstorvolumemanager.py (hunk 1 above)
Comment (Wescoeur) [reply drivers/linstorvolumemanager.py]: It's normally a `break`. I fix it. :)

File: drivers/LinstorSR.py (hunk 2 above)
Comment (Wescoeur) [reply drivers/LinstorSR.py]: Yeah it's not very elegant... The names are already defined like this in the other drivers, for the moment I'm not sure it's a good idea to change that.

Comment (benjamreis) [reply drivers/LinstorSR.py]: Indeed let's not introduce too much differences in this PR (not its point).
Perhaps we'll unify that later in another one. :)

**Comment by Wescoeur** [comment]
I close this useless PR, all changes are now in https://github.com/xcp-ng/sm/pull/2.

**Pull request closed without merging**